src/itemButtonList.js

```diff
@@ -0,0 +1,220 @@
+        if (itemId) {
+            const $target = component.getElement().find(cssSelectors.itemById(itemId));
+            if ($target.length) {
+                $target.addClass(cssClasses.active);
+                // finally make sure the item is visible
+                autoscroll($target, component.getElement());
+            }
+            const $ariaTarget = component.getElement().find(cssSelectors.navigableById(itemId));
+            if ($ariaTarget.length) {
+                $ariaTarget.attr('aria-current', 'location');
+            }
+        }
+    };
+
+    /**
+     * 'tabfocus' styling, for Safari until :focus-visible supported
+     * @param {jQuery|null}  $target
+     */
+    const setFocusStyle = $target => {
+        component.getElement()
+            .find(cssSelectors.keyfocused)
+            .removeClass(cssClasses.keyfocused);
+
+        if ($target && $target.length) {
+            $target.addClass(cssClasses.keyfocused);
+        }
+    };
+
+    /**
+     * Apply a callback on each navigable element
+     * @param {*} callback
+     */
+    const eachNavigable = callback => {
+        component.getElement()
+            .find(cssSelectors.navigable)
+            .each(callback);
+    };
+
+    /**
+     * Enables the keyboard navigation using 'tab' keys
+     */
+    const enableKeyboard = () => {
+        eachNavigable((index, el) => el.removeAttribute('tabindex'));
+    };
+
+    /**
+     * Disables the keyboard navigation using 'tab' keys
+     */
+    const disableKeyboard = () => {
+        eachNavigable((index, el) => el.setAttribute('tabindex', -1));
+    };
+
+    /**
+     * Emits the click event detailing the clicked item
+     * The active item change should be handled by the consumer through the API, in case it is conditional or asynchronous
+     * @param {String} itemId
+     */
+    const onClick = (itemId) => {
+        /**
+         * @event click
+         * @param {String} itemId
+         * @param {Number} position
+         */
+        component.trigger('click', { id: itemId });
+    };
+
+    /**
+     * Defines the buttonList API
+     * @type {buttonList}
+     */
+    const api = {
+        /**
+         * Sets the active item
+         * @param {String} itemId
+         * @returns {buttonList}
+         */
+        setActiveItem(itemId) {
+            activeItemId = itemId;
+            if (this.is('rendered')) {
+                selectItem(itemId);
+            }
+            return this;
+        }
+    };
+
+    /**
+     * @typedef {component} buttonList
+     */
+    component = componentFactory(api, {})
+        // set the component's layout
+        .setTemplate(itemButtonListTpl)
+        // renders the component
+        .on('render', function onItemButtonListRender() {
+            // 'tabfocus' detection, for Safari until :focus-visible supported
+            this.getElement().on('keydown', cssSelectors.navigable, e => {
+                if (e.key === 'Tab') {
```

Comment:
~~Note 2: I couldn't test it in Safari (browserstack vs local lti-launch problems), but looks like you did and this detection works there.~~
Checked in TAE & Browserstack Safari 15.1 - doesn't seem to work.

Reply:
I do not know about proper key navigation code. I haven't seen any arrow key navigation working in the taoQtiTest fizzy panel. I guess we will align it later if it's not right.

You are right about Safari - I will investigate.

Reply:
About arrow key - you should first click on some step button  with mouse (=focus it), then immediately after that use arrow keys. But yes, it counts as not-working)

Reply:
@svoronuk But I think we don't use native behavior, we use focus-visible polyfill package. Or what do you mean?

Reply:
@svoronuk This is why I've added `:focus-visible` styles to this component. It will soon be supported in Safari and we won't need a polyfill - but we are not there yet. So I left in the code to toggle `buttonlist-item-focus` classes on Tab.

It doesn't handle arrow keys at all yet, and I don't plan to add that in the current task.

Reply:
@olga-kulish i just mean that our accessibility plugin is huge and complex which uses various focusing strategies and managers. All this code is focused on aligning behavior through the browsers for various parts on test runner. In case of panel i prefer avoid using this plugin and implement keyboard focusing as simple as possible)

Reply:
@oatymart arrow key navigation is working for fizzy panel in test. You just need focus panel overall, than you can start using arrows.

Reply:
** thought**: Indeed, the accessibility plugin will take care of the keyboard navigation in the context of the test runner. But the reason why we have such a plugin is not because of IE support. Actually, it was implemented to prevent the user from navigating away from the test while taking it. The idea was to give a consistent experience, across browser, and at the same time control what elements the user was able to navigate through.

Anyway, I think this should not be taken into account here. This package is meant to be generic, and it should not bother about what consumer will use it and how.

**question:** This bring me to a question: is it relevant to manage explicitly the keyboard navigation here? Should not it be leaved to the native behaviour by only setting the proper tabstops attributes?

Reply:
> Should not it be leaved to the native behaviour by only setting the proper tabstops attributes?

Ideally yes. But this approach brings one other feature - applying focused styling only when the keyboard is used (like `:focus-visible`). If you think this is not suitable in CurrentGen or needed on the review side - I will happily delete these listeners.